mantidproject/mantid
# Add 30 group texture option to EngDiff UI

**Description of work.**

Defined a grouping .xml that splits the EnginX detectors into 30 groups (15 per bank).
Have added the option to use this grouping in the engineering diffraction UI

**To test:**
(0) Open EngDiff UI (Interfaces > Diffraction > Engineerign Diffraction and type `texture30_PR` in the RB box at the top
(1) In the calibration tab type run number 193749 in the calibration sample box
(2) Check Plot Calibrated Workspace and Crop Calibration and select Texture30 from the drop-down box that appears
(3) Click the Calibrate button - when finished it should open up plot windows with a combined total of 30 subplot axes.
(4) Check the workspace `focussed ceria` has 30 spectra
(5) Check that there are two files at `save_dir\User\texture30_PR\Calibration\ENGINX_193749_Texture30.prm` & `.nxs` where `save_dir` is displayed at bottom right of UI window (and can be set in settings, cog button in bottom left of UI window). 
(5) Go on to the Focus tab and check there is some text saying `Chosen Region of Interest: Texture30` below the vanadium run number box.
(6) Type Sample Run = 193749 (same ceria run) and Vanadium = 236516
(7) Click Focus button
(8) Check that there are 30 spectra in the focussed output workspace `193749_engggui_focusing_output_ws_Texture30`
(9) Check that in the directory  `save_dir\User\texture30_PR\Calibration\` there are

- 2 .abc files (TOF and dSpacing)
- 2 .gss file (TOF and dSpacing)
- 30 files ending `TOF.nxs` 
- 30 files ending `dSpacing.nxs` 

(10) Clear the ADS, go back to the calibration tab and click load existing calibration
(11) Browse to the .prm created `save_dir\User\texture30_PR\Calibration\ENGINX_193749_Texture30.prm`
(12) Repeat steps (2-9)

(13) Run the systems test like so (having opened terminal using command_prompt.bat in build folder)
`systemtest.bat -C debug -R EnginXScriptTest.FocusTexture30 -j 2`

Fixes #32836 





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
